From e01145a8fd7be3122287dc5316436c547e8ae932 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Sun, 24 Nov 2024 13:28:07 +0200
Subject: [PATCH] [FSW-1317] fs1: add gt ip rl dts entries

This patch adds dts entries 160 CU_RRTR IPs that will be used to rate
limit the GT IP. These will be rate limited in mgmt application
running on compute die and will use compute die dup engine region 1
for RL configurations.

Change-Id: I263a2f87da1270313cca506fb3e95579702fe846
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/dts/arc/fs1_emb.dtsi b/dts/arc/fs1_emb.dtsi
index 8cfeac2..2caebe7 100644
--- a/dts/arc/fs1_emb.dtsi
+++ b/dts/arc/fs1_emb.dtsi
@@ -155,6 +155,356 @@
 		label = "PDMA_COMPUTE_RL";
 	};
 
+	clk_rl_cu0_rrtr_compute: clk-rl-cu0-rrtr-compute@f1a00ae0 {
+		status = "disabled";
+		compatible = "hl,rl";
+		reg = <0xf1a00ae0 0x10>,
+		      <0xf1a01ae0 0x10>,
+		      <0xf1a02ae0 0x10>,
+		      <0xf1a03ae0 0x10>,
+		      <0xf1a04ae0 0x10>,
+		      <0xf1a10ae0 0x10>,
+		      <0xf1a11ae0 0x10>,
+		      <0xf1a12ae0 0x10>,
+		      <0xf1a13ae0 0x10>,
+		      <0xf1a14ae0 0x10>,
+		      <0xf1a20ae0 0x10>,
+		      <0xf1a21ae0 0x10>,
+		      <0xf1a22ae0 0x10>,
+		      <0xf1a23ae0 0x10>,
+		      <0xf1a24ae0 0x10>,
+		      <0xf1a30ae0 0x10>,
+		      <0xf1a31ae0 0x10>,
+		      <0xf1a32ae0 0x10>,
+		      <0xf1a33ae0 0x10>,
+		      <0xf1a34ae0 0x10>,
+		      <0xf1a40ae0 0x10>,
+		      <0xf1a41ae0 0x10>,
+		      <0xf1a42ae0 0x10>,
+		      <0xf1a43ae0 0x10>,
+		      <0xf1a44ae0 0x10>,
+		      <0xf1a50ae0 0x10>,
+		      <0xf1a51ae0 0x10>,
+		      <0xf1a52ae0 0x10>,
+		      <0xf1a53ae0 0x10>,
+		      <0xf1a54ae0 0x10>,
+		      <0xf1a60ae0 0x10>,
+		      <0xf1a61ae0 0x10>,
+		      <0xf1a62ae0 0x10>,
+		      <0xf1a63ae0 0x10>,
+		      <0xf1a64ae0 0x10>,
+		      <0xf1a70ae0 0x10>,
+		      <0xf1a71ae0 0x10>,
+		      <0xf1a72ae0 0x10>,
+		      <0xf1a73ae0 0x10>,
+		      <0xf1a74ae0 0x10>,
+		      <0xf1a80ae0 0x10>,
+		      <0xf1a81ae0 0x10>,
+		      <0xf1a82ae0 0x10>,
+		      <0xf1a83ae0 0x10>,
+		      <0xf1a84ae0 0x10>,
+		      <0xf1a90ae0 0x10>,
+		      <0xf1a91ae0 0x10>,
+		      <0xf1a92ae0 0x10>,
+		      <0xf1a93ae0 0x10>,
+		      <0xf1a94ae0 0x10>,
+		      <0xf1aa0ae0 0x10>,
+		      <0xf1aa1ae0 0x10>,
+		      <0xf1aa2ae0 0x10>,
+		      <0xf1aa3ae0 0x10>,
+		      <0xf1aa4ae0 0x10>,
+		      <0xf1ab0ae0 0x10>,
+		      <0xf1ab1ae0 0x10>,
+		      <0xf1ab2ae0 0x10>,
+		      <0xf1ab3ae0 0x10>,
+		      <0xf1ab4ae0 0x10>,
+		      <0xf1ac0ae0 0x10>,
+		      <0xf1ac1ae0 0x10>,
+		      <0xf1ac2ae0 0x10>,
+		      <0xf1ac3ae0 0x10>,
+		      <0xf1ac4ae0 0x10>,
+		      <0xf1ad0ae0 0x10>,
+		      <0xf1ad1ae0 0x10>,
+		      <0xf1ad2ae0 0x10>,
+		      <0xf1ad3ae0 0x10>,
+		      <0xf1ad4ae0 0x10>,
+		      <0xf1ae0ae0 0x10>,
+		      <0xf1ae1ae0 0x10>,
+		      <0xf1ae2ae0 0x10>,
+		      <0xf1ae3ae0 0x10>,
+		      <0xf1ae4ae0 0x10>,
+		      <0xf1af0ae0 0x10>,
+		      <0xf1af1ae0 0x10>,
+		      <0xf1af2ae0 0x10>,
+		      <0xf1af3ae0 0x10>,
+		      <0xf1af4ae0 0x10>;
+		reg-names = "compute_cu0_rrtr_0_rtr_ctrl_0",
+			    "compute_cu0_rrtr_0_rtr_ctrl_1",
+			    "compute_cu0_rrtr_0_rtr_ctrl_2",
+			    "compute_cu0_rrtr_0_rtr_ctrl_3",
+			    "compute_cu0_rrtr_0_rtr_ctrl_4",
+			    "compute_cu0_rrtr_1_rtr_ctrl_0",
+			    "compute_cu0_rrtr_1_rtr_ctrl_1",
+			    "compute_cu0_rrtr_1_rtr_ctrl_2",
+			    "compute_cu0_rrtr_1_rtr_ctrl_3",
+			    "compute_cu0_rrtr_1_rtr_ctrl_4",
+			    "compute_cu0_rrtr_2_rtr_ctrl_0",
+			    "compute_cu0_rrtr_2_rtr_ctrl_1",
+			    "compute_cu0_rrtr_2_rtr_ctrl_2",
+			    "compute_cu0_rrtr_2_rtr_ctrl_3",
+			    "compute_cu0_rrtr_2_rtr_ctrl_4",
+			    "compute_cu0_rrtr_3_rtr_ctrl_0",
+			    "compute_cu0_rrtr_3_rtr_ctrl_1",
+			    "compute_cu0_rrtr_3_rtr_ctrl_2",
+			    "compute_cu0_rrtr_3_rtr_ctrl_3",
+			    "compute_cu0_rrtr_3_rtr_ctrl_4",
+			    "compute_cu0_rrtr_4_rtr_ctrl_0",
+			    "compute_cu0_rrtr_4_rtr_ctrl_1",
+			    "compute_cu0_rrtr_4_rtr_ctrl_2",
+			    "compute_cu0_rrtr_4_rtr_ctrl_3",
+			    "compute_cu0_rrtr_4_rtr_ctrl_4",
+			    "compute_cu0_rrtr_5_rtr_ctrl_0",
+			    "compute_cu0_rrtr_5_rtr_ctrl_1",
+			    "compute_cu0_rrtr_5_rtr_ctrl_2",
+			    "compute_cu0_rrtr_5_rtr_ctrl_3",
+			    "compute_cu0_rrtr_5_rtr_ctrl_4",
+			    "compute_cu0_rrtr_6_rtr_ctrl_0",
+			    "compute_cu0_rrtr_6_rtr_ctrl_1",
+			    "compute_cu0_rrtr_6_rtr_ctrl_2",
+			    "compute_cu0_rrtr_6_rtr_ctrl_3",
+			    "compute_cu0_rrtr_6_rtr_ctrl_4",
+			    "compute_cu0_rrtr_7_rtr_ctrl_0",
+			    "compute_cu0_rrtr_7_rtr_ctrl_1",
+			    "compute_cu0_rrtr_7_rtr_ctrl_2",
+			    "compute_cu0_rrtr_7_rtr_ctrl_3",
+			    "compute_cu0_rrtr_7_rtr_ctrl_4",
+			    "compute_cu0_rrtr_8_rtr_ctrl_0",
+			    "compute_cu0_rrtr_8_rtr_ctrl_1",
+			    "compute_cu0_rrtr_8_rtr_ctrl_2",
+			    "compute_cu0_rrtr_8_rtr_ctrl_3",
+			    "compute_cu0_rrtr_8_rtr_ctrl_4",
+			    "compute_cu0_rrtr_9_rtr_ctrl_0",
+			    "compute_cu0_rrtr_9_rtr_ctrl_1",
+			    "compute_cu0_rrtr_9_rtr_ctrl_2",
+			    "compute_cu0_rrtr_9_rtr_ctrl_3",
+			    "compute_cu0_rrtr_9_rtr_ctrl_4",
+			    "compute_cu0_rrtr_10_rtr_ctrl_0",
+			    "compute_cu0_rrtr_10_rtr_ctrl_1",
+			    "compute_cu0_rrtr_10_rtr_ctrl_2",
+			    "compute_cu0_rrtr_10_rtr_ctrl_3",
+			    "compute_cu0_rrtr_10_rtr_ctrl_4",
+			    "compute_cu0_rrtr_11_rtr_ctrl_0",
+			    "compute_cu0_rrtr_11_rtr_ctrl_1",
+			    "compute_cu0_rrtr_11_rtr_ctrl_2",
+			    "compute_cu0_rrtr_11_rtr_ctrl_3",
+			    "compute_cu0_rrtr_11_rtr_ctrl_4",
+			    "compute_cu0_rrtr_12_rtr_ctrl_0",
+			    "compute_cu0_rrtr_12_rtr_ctrl_1",
+			    "compute_cu0_rrtr_12_rtr_ctrl_2",
+			    "compute_cu0_rrtr_12_rtr_ctrl_3",
+			    "compute_cu0_rrtr_12_rtr_ctrl_4",
+			    "compute_cu0_rrtr_13_rtr_ctrl_0",
+			    "compute_cu0_rrtr_13_rtr_ctrl_1",
+			    "compute_cu0_rrtr_13_rtr_ctrl_2",
+			    "compute_cu0_rrtr_13_rtr_ctrl_3",
+			    "compute_cu0_rrtr_13_rtr_ctrl_4",
+			    "compute_cu0_rrtr_14_rtr_ctrl_0",
+			    "compute_cu0_rrtr_14_rtr_ctrl_1",
+			    "compute_cu0_rrtr_14_rtr_ctrl_2",
+			    "compute_cu0_rrtr_14_rtr_ctrl_3",
+			    "compute_cu0_rrtr_14_rtr_ctrl_4",
+			    "compute_cu0_rrtr_15_rtr_ctrl_0",
+			    "compute_cu0_rrtr_15_rtr_ctrl_1",
+			    "compute_cu0_rrtr_15_rtr_ctrl_2",
+			    "compute_cu0_rrtr_15_rtr_ctrl_3",
+			    "compute_cu0_rrtr_15_rtr_ctrl_4";
+		#clock-cells = <1>;
+		clocks = <&refclk>;
+		max_sat = <20>;
+		block_type = "mstr_if";
+		dup_eng_inst = <&dup_eng1_compute>;
+		dup_grp_id = <RL_GROUP_ID_CU0_RRTR_RL_GRP_0
+			    RL_GROUP_ID_CU0_RRTR_RL_GRP_1
+			    RL_GROUP_ID_CU0_RRTR_RL_GRP_2>;
+		dup_master_grp_id = <RL_GROUP_ID_CU0_RRTR_RL_MSTR_GRP>;
+		label = "CU0_RRTR_RL";
+	};
+
+	clk_rl_cu1_rrtr_compute: clk-rl-cu1-rrtr-compute@f1b00ae0 {
+		status = "disabled";
+		compatible = "hl,rl";
+		reg = <0xf1b00ae0 0x10>,
+		      <0xf1b01ae0 0x10>,
+		      <0xf1b02ae0 0x10>,
+		      <0xf1b03ae0 0x10>,
+		      <0xf1b04ae0 0x10>,
+		      <0xf1b10ae0 0x10>,
+		      <0xf1b11ae0 0x10>,
+		      <0xf1b12ae0 0x10>,
+		      <0xf1b13ae0 0x10>,
+		      <0xf1b14ae0 0x10>,
+		      <0xf1b20ae0 0x10>,
+		      <0xf1b21ae0 0x10>,
+		      <0xf1b22ae0 0x10>,
+		      <0xf1b23ae0 0x10>,
+		      <0xf1b24ae0 0x10>,
+		      <0xf1b30ae0 0x10>,
+		      <0xf1b31ae0 0x10>,
+		      <0xf1b32ae0 0x10>,
+		      <0xf1b33ae0 0x10>,
+		      <0xf1b34ae0 0x10>,
+		      <0xf1b40ae0 0x10>,
+		      <0xf1b41ae0 0x10>,
+		      <0xf1b42ae0 0x10>,
+		      <0xf1b43ae0 0x10>,
+		      <0xf1b44ae0 0x10>,
+		      <0xf1b50ae0 0x10>,
+		      <0xf1b51ae0 0x10>,
+		      <0xf1b52ae0 0x10>,
+		      <0xf1b53ae0 0x10>,
+		      <0xf1b54ae0 0x10>,
+		      <0xf1b60ae0 0x10>,
+		      <0xf1b61ae0 0x10>,
+		      <0xf1b62ae0 0x10>,
+		      <0xf1b63ae0 0x10>,
+		      <0xf1b64ae0 0x10>,
+		      <0xf1b70ae0 0x10>,
+		      <0xf1b71ae0 0x10>,
+		      <0xf1b72ae0 0x10>,
+		      <0xf1b73ae0 0x10>,
+		      <0xf1b74ae0 0x10>,
+		      <0xf1b80ae0 0x10>,
+		      <0xf1b81ae0 0x10>,
+		      <0xf1b82ae0 0x10>,
+		      <0xf1b83ae0 0x10>,
+		      <0xf1b84ae0 0x10>,
+		      <0xf1b90ae0 0x10>,
+		      <0xf1b91ae0 0x10>,
+		      <0xf1b92ae0 0x10>,
+		      <0xf1b93ae0 0x10>,
+		      <0xf1b94ae0 0x10>,
+		      <0xf1ba0ae0 0x10>,
+		      <0xf1ba1ae0 0x10>,
+		      <0xf1ba2ae0 0x10>,
+		      <0xf1ba3ae0 0x10>,
+		      <0xf1ba4ae0 0x10>,
+		      <0xf1bb0ae0 0x10>,
+		      <0xf1bb1ae0 0x10>,
+		      <0xf1bb2ae0 0x10>,
+		      <0xf1bb3ae0 0x10>,
+		      <0xf1bb4ae0 0x10>,
+		      <0xf1bc0ae0 0x10>,
+		      <0xf1bc1ae0 0x10>,
+		      <0xf1bc2ae0 0x10>,
+		      <0xf1bc3ae0 0x10>,
+		      <0xf1bc4ae0 0x10>,
+		      <0xf1bd0ae0 0x10>,
+		      <0xf1bd1ae0 0x10>,
+		      <0xf1bd2ae0 0x10>,
+		      <0xf1bd3ae0 0x10>,
+		      <0xf1bd4ae0 0x10>,
+		      <0xf1be0ae0 0x10>,
+		      <0xf1be1ae0 0x10>,
+		      <0xf1be2ae0 0x10>,
+		      <0xf1be3ae0 0x10>,
+		      <0xf1be4ae0 0x10>,
+		      <0xf1bf0ae0 0x10>,
+		      <0xf1bf1ae0 0x10>,
+		      <0xf1bf2ae0 0x10>,
+		      <0xf1bf3ae0 0x10>,
+		      <0xf1bf4ae0 0x10>;
+		reg-names = "compute_cu1_rrtr_0_rtr_ctrl_0",
+			    "compute_cu1_rrtr_0_rtr_ctrl_1",
+			    "compute_cu1_rrtr_0_rtr_ctrl_2",
+			    "compute_cu1_rrtr_0_rtr_ctrl_3",
+			    "compute_cu1_rrtr_0_rtr_ctrl_4",
+			    "compute_cu1_rrtr_1_rtr_ctrl_0",
+			    "compute_cu1_rrtr_1_rtr_ctrl_1",
+			    "compute_cu1_rrtr_1_rtr_ctrl_2",
+			    "compute_cu1_rrtr_1_rtr_ctrl_3",
+			    "compute_cu1_rrtr_1_rtr_ctrl_4",
+			    "compute_cu1_rrtr_2_rtr_ctrl_0",
+			    "compute_cu1_rrtr_2_rtr_ctrl_1",
+			    "compute_cu1_rrtr_2_rtr_ctrl_2",
+			    "compute_cu1_rrtr_2_rtr_ctrl_3",
+			    "compute_cu1_rrtr_2_rtr_ctrl_4",
+			    "compute_cu1_rrtr_3_rtr_ctrl_0",
+			    "compute_cu1_rrtr_3_rtr_ctrl_1",
+			    "compute_cu1_rrtr_3_rtr_ctrl_2",
+			    "compute_cu1_rrtr_3_rtr_ctrl_3",
+			    "compute_cu1_rrtr_3_rtr_ctrl_4",
+			    "compute_cu1_rrtr_4_rtr_ctrl_0",
+			    "compute_cu1_rrtr_4_rtr_ctrl_1",
+			    "compute_cu1_rrtr_4_rtr_ctrl_2",
+			    "compute_cu1_rrtr_4_rtr_ctrl_3",
+			    "compute_cu1_rrtr_4_rtr_ctrl_4",
+			    "compute_cu1_rrtr_5_rtr_ctrl_0",
+			    "compute_cu1_rrtr_5_rtr_ctrl_1",
+			    "compute_cu1_rrtr_5_rtr_ctrl_2",
+			    "compute_cu1_rrtr_5_rtr_ctrl_3",
+			    "compute_cu1_rrtr_5_rtr_ctrl_4",
+			    "compute_cu1_rrtr_6_rtr_ctrl_0",
+			    "compute_cu1_rrtr_6_rtr_ctrl_1",
+			    "compute_cu1_rrtr_6_rtr_ctrl_2",
+			    "compute_cu1_rrtr_6_rtr_ctrl_3",
+			    "compute_cu1_rrtr_6_rtr_ctrl_4",
+			    "compute_cu1_rrtr_7_rtr_ctrl_0",
+			    "compute_cu1_rrtr_7_rtr_ctrl_1",
+			    "compute_cu1_rrtr_7_rtr_ctrl_2",
+			    "compute_cu1_rrtr_7_rtr_ctrl_3",
+			    "compute_cu1_rrtr_7_rtr_ctrl_4",
+			    "compute_cu1_rrtr_8_rtr_ctrl_0",
+			    "compute_cu1_rrtr_8_rtr_ctrl_1",
+			    "compute_cu1_rrtr_8_rtr_ctrl_2",
+			    "compute_cu1_rrtr_8_rtr_ctrl_3",
+			    "compute_cu1_rrtr_8_rtr_ctrl_4",
+			    "compute_cu1_rrtr_9_rtr_ctrl_0",
+			    "compute_cu1_rrtr_9_rtr_ctrl_1",
+			    "compute_cu1_rrtr_9_rtr_ctrl_2",
+			    "compute_cu1_rrtr_9_rtr_ctrl_3",
+			    "compute_cu1_rrtr_9_rtr_ctrl_4",
+			    "compute_cu1_rrtr_10_rtr_ctrl_0",
+			    "compute_cu1_rrtr_10_rtr_ctrl_1",
+			    "compute_cu1_rrtr_10_rtr_ctrl_2",
+			    "compute_cu1_rrtr_10_rtr_ctrl_3",
+			    "compute_cu1_rrtr_10_rtr_ctrl_4",
+			    "compute_cu1_rrtr_11_rtr_ctrl_0",
+			    "compute_cu1_rrtr_11_rtr_ctrl_1",
+			    "compute_cu1_rrtr_11_rtr_ctrl_2",
+			    "compute_cu1_rrtr_11_rtr_ctrl_3",
+			    "compute_cu1_rrtr_11_rtr_ctrl_4",
+			    "compute_cu1_rrtr_12_rtr_ctrl_0",
+			    "compute_cu1_rrtr_12_rtr_ctrl_1",
+			    "compute_cu1_rrtr_12_rtr_ctrl_2",
+			    "compute_cu1_rrtr_12_rtr_ctrl_3",
+			    "compute_cu1_rrtr_12_rtr_ctrl_4",
+			    "compute_cu1_rrtr_13_rtr_ctrl_0",
+			    "compute_cu1_rrtr_13_rtr_ctrl_1",
+			    "compute_cu1_rrtr_13_rtr_ctrl_2",
+			    "compute_cu1_rrtr_13_rtr_ctrl_3",
+			    "compute_cu1_rrtr_13_rtr_ctrl_4",
+			    "compute_cu1_rrtr_14_rtr_ctrl_0",
+			    "compute_cu1_rrtr_14_rtr_ctrl_1",
+			    "compute_cu1_rrtr_14_rtr_ctrl_2",
+			    "compute_cu1_rrtr_14_rtr_ctrl_3",
+			    "compute_cu1_rrtr_14_rtr_ctrl_4",
+			    "compute_cu1_rrtr_15_rtr_ctrl_0",
+			    "compute_cu1_rrtr_15_rtr_ctrl_1",
+			    "compute_cu1_rrtr_15_rtr_ctrl_2",
+			    "compute_cu1_rrtr_15_rtr_ctrl_3",
+			    "compute_cu1_rrtr_15_rtr_ctrl_4";
+		#clock-cells = <1>;
+		clocks = <&refclk>;
+		max_sat = <20>;
+		block_type = "mstr_if";
+		dup_eng_inst = <&dup_eng1_compute>;
+		dup_grp_id = <RL_GROUP_ID_CU1_RRTR_RL_GRP_0
+			    RL_GROUP_ID_CU1_RRTR_RL_GRP_1
+			    RL_GROUP_ID_CU1_RRTR_RL_GRP_2>;
+		dup_master_grp_id = <RL_GROUP_ID_CU1_RRTR_RL_MSTR_GRP>;
+		label = "CU1_RRTR_RL";
+	};
+
 	clk_d2d_compute: clk-d2d-compute@f0348000 {
 		status = "disabled";
 		compatible = "hl,pll";
